extensions/amp-youtube/1.0/youtube.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {VideoEvents} from '../../../src/video-interface';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+import {VideoWrapper} from '../../amp-video/1.0/video-wrapper';
+import {addParamsToUrl} from '../../../src/url';
+import {dict} from '../../../src/utils/object';
+import {mutedOrUnmutedEvent, objOrParseJson} from '../../../src/iframe-video';
+import {useState} from '../../../src/preact';
+
+// Correct PlayerStates taken from
+// https://developers.google.com/youtube/iframe_api_reference#Playback_status
+/**
+ * @enum {string}
+ * @private
+ */
+const PlayerStates = {
+  '-1': 'unstarted',
+  '0': 'ended',
+  '1': 'playing',
+  '2': 'paused',
+  '3': 'buffering',
+  '5': 'video_cued',
+};
+
+/**
+ * @enum {string}
+ * @private
+ */
+const methods = {
+  'play': 'playVideo',
+  'pause': 'pauseVideo',
+  'unmute': 'unMute',
+};
+
+/**
+ * @enum {number}
+ * @private
+ */
+const PlayerFlags = {
+  // Config to tell YouTube to hide annotations by default
+  HIDE_ANNOTATION: 3,
+};
+
+/**
+ * @param {!YoutubeProps} props
+ * @return {PreactDef.Renderable}
+ */
+export function Youtube({
+  autoplay,
+  loop,
+  videoid,
+  liveChannelid,
+  params = {},
+  // eslint-disable-next-line no-unused-vars
+  dock, // TODO: dock is unused... how is it meant to be used?
```

Comment:
@alanorozco Will docking be handled by VideoWrapper in any way? It's not mentioned in the old `amp-youtube`.